firefly_client/firefly_client.py

```diff
@@ -1389,14 +1391,67 @@ def set_stretch(self, plot_id, stype=None, algorithm=None, **additional_params):
         elif stype and (stype.lower() in ['minmax', 'maxmin']):
             # 'maxmin' retained for backwards compatibility
             serialized_rv = self._create_rangevalues_standard(algorithm, 'percent', 
-                                                              lower_value=0, upper_value=100)
+                                                              lower_value=0, upper_value=100, **additional_params)
         else:
             serialized_rv = self._create_rangevalues_standard(algorithm, stype, **additional_params)
 
         st_data = [{'band': 'NO_BAND', 'rv': serialized_rv, 'bandVisible': True}]
         payload = {'stretchData': st_data, 'plotId': plot_id}
 
-        return self.dispatch_remote_action_by_post(self.channel, FireflyClient.ACTION_DICT['StretchImage'], payload)
+        return_val = self.dispatch_remote_action_by_post(self.channel, FireflyClient.ACTION_DICT['StretchImage'], payload)
+        return_val['rv_string'] = serialized_rv
+        return return_val
+
+    def parse_rvstring(self, rvstring):
+        """parse a Firefly RangeValues string into a dictionary
+
+        Parameters:
+        -----------
+        rvstring : `str`
+            RangeValues string as returned by the set_stretch method.
+
+        Returns:
+        --------
+        outdict : `dict`
+            dictionary of the inputs
+        """
+        vals = rvstring.split(',')
+        assert len(vals) == 10
+        outdict = dict(lower_type = self.INVERSE_STRETCH_TYPE[int(vals[0])],
+                   lower_limit = float(vals[1]),
+                   upper_type = self.INVERSE_STRETCH_TYPE[int(vals[2])],
+                   upper_limit = float(vals[3]),
+                   asinh_q = float(vals[4]),
+                   gamma = float(vals[5]),
```

Comment:
Do we want to keep key names the same as in set_stretch? Then we don't need documentation, and it makes it easier for users to use the dictionary with set_stretch. But I don't have a strong opinion about it.

Reply:
I suppose this reflects my desire to shorten `asinh_q_value` and `gamma_value` -- but that should be done in `set_stretch`. It's better to match with `set_stretch` for now.

I also realized that in general one can have different STRETCH_TYPE for upper and lower values, but `set_stretch` doesn't support that (nor does `lsst.afw.display`).